src/network/local_socket_reply.cpp

```diff
@@ -141,36 +139,113 @@ void mp::LocalSocketReply::send_request(const QNetworkRequest& request, QIODevic
     // Build the HTTP method part
     http_data += op;
     http_data += ' ';
-    http_data += request.url().toString();
+    http_data += request.url().path();
+
+    if (request.url().hasQuery())
+    {
+        http_data += "?";
+        http_data += request.url().query();
+    }
+
     http_data += " HTTP/1.1\r\n";
 
     // Build the HTTP Host header
-    // Host can be anything, so we'll use 'multipass'
-    http_data += "Host: multipass\r\n";
+    auto host = request.url().host();
+    if (!host.isEmpty())
+    {
+        http_data += "Host: ";
+        http_data += host;
+        http_data += "\r\n";
```

Comment:
If you can think of a more concise way of doing this, could be nice. There's a _lot_ of `… += "\r\n"` below…

Reply:
Well, this is how Qt does it in their network related code, so I'm following their style.  That said, I can use `fmt::format()` for building the string and then use `QByteArray::fromStdString()` to convert it.  Or a `QString().arg()` and `QString.toUtf8()` combo.

Reply:
Your call if you think it worth it at all.

Reply:
I kept the `QByteArray`'s, but I tightened up the construction of the `char` data and should look much better now.